R/slice.R

```diff
@@ -325,56 +325,43 @@ slice_eval <- function(mask, dots, error_call = caller_env()) {
   )
 }
 
-slice_combine <- function(chunks, mask, error_call = caller_env()) {
+slice_combine <- function(chunks, dots, mask, error_call = caller_env()) {
   rows <- mask$get_rows()
   slice_indices <- new_list(length(rows))
 
   withCallingHandlers(
     for (group in seq_along(rows)) {
       current_rows <- rows[[group]]
-      res <- chunks[[group]]
-
-      if (is.logical(res) && all(is.na(res))) {
-        res <- integer()
-      } else if (is.numeric(res)) {
-        if (is.matrix(res) && ncol(res) == 1) {
-          res <- as.vector(res)
-        }
-        res <- fix_call(vec_cast(res, integer(), x_arg = ""), NULL)
-      } else {
-        bullets <- c(
-          glue("Invalid result of type <{vec_ptype_full(res)}>."),
-          i = "Indices must be positive or negative integers."
-        )
-        abort(bullets, call = NULL)
-      }
 
-      if (length(res) == 0L) {
-        # nothing to do
-      } else if (all(res >= 0, na.rm = TRUE)) {
-        res <- res[!is.na(res) & res <= length(current_rows) & res > 0]
-      } else if (all(res <= 0, na.rm = TRUE)) {
-        res <- setdiff(seq_along(current_rows), -res)
-      } else {
-        mask$set_current_group(group)
-        n_positive <- sum(res >= 0, na.rm = TRUE)
-        n_negative <- sum(res <= 0, na.rm = TRUE)
-        bullets <- c(
-          "Indices must be all positive or all negative.",
-          i = glue("Got {n_positive} positives, {n_negative} negatives.")
-        )
-        abort(bullets, call = NULL)
-      }
+      i <- vec_as_subscript(
+        chunks[[group]],
+        logical = "error",
+        character = "error",
+        arg = as_label(dots[[group]]),
+        call = NULL
```

Comment:
Should this be:

```suggestion
        call = error_call
```

(also below)

Reply:
Ah no because these errors are chained to a `slice()` error. Maybe mention this in a comment since this puzzled me?